ownCloud/Viewer/WebViewDisplayViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  ImageDisplayViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 30/08/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+import WebKit
+
+class WebViewDisplayViewController: DisplayViewController, DisplayViewProtocol {
+	static var supportedMimeTypes: [String] =
+		["image/jpeg",
+		 "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+		 "application/vnd.openxmlformats-officedocument.wordprocessingml.document"]
+
+	static var features: [String : Any]? = [FeatureKeys.canEdit : true]
+
+	static func webViewExtension(identifier: String) -> OCExtension {
+		let rawIdentifier: OCExtensionIdentifier =  OCExtensionIdentifier(rawValue: identifier)
+		var locationIdentifiers: [OCExtensionLocationIdentifier] = []
+		for mimeType in supportedMimeTypes {
+			let locationIdentifier: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier(rawValue: mimeType)
+			locationIdentifiers.append(locationIdentifier)
+		}
```

Comment:
Given that OCExtensionLocationIdentifiers are also just Strings… do you know of a more direct way to cast this? I couldn't immediately find any.

Reply:
I've been looking for some other option and I found some info about how `NS_TYPED_EXTENSIBLE_ENUM` is imported into Swift: [Declare Typed Extensible Enumerations](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/grouping_related_objective-c_constants)

The solution they propose is to create an extension with a static computed variable for each new case you want to add. Something like:

```Swift
extension OCExtensionLocationIdentifier {
    static var pdf: OCExtensionLocationIdentifier {
        return OCExtensionLocationIdentifier(rawValue: "application/pdf") 
    }
}
```

I think this is not extensible for our use case because we could have a lot of mime types.

I coded the `OCExtension` creation in a `DisplayExtension` extension that provides the default implementation, so this code is only written once. It could be overridden in the future by any class that implements this protocol.

```Swift
static var displayExtension: OCExtension {
		let rawIdentifier: OCExtensionIdentifier =  OCExtensionIdentifier(rawValue: displayExtensionIdentifier)
		var locationIdentifiers: [OCExtensionLocationIdentifier] = []
		for mimeType in supportedMimeTypes {
			let locationIdentifier: OCExtensionLocationIdentifier = OCExtensionLocationIdentifier(rawValue: mimeType)
			locationIdentifiers.append(locationIdentifier)
		}

		let displayExtension = OCExtension(identifier: rawIdentifier, type: .viewer, locations: locationIdentifiers, features: features) { (_ rawExtension, _ context, _ error) -> Any? in
			return Self()
		}

		return displayExtension!
	}
}
```